### PR TITLE
Remove `_coerce_transformlike_arg` and use `validate_transform4x4` instead

### PR DESCRIPTION
**Pull request by user27182**
### Overview
Follow-up to #6417.

Replace `_coerce_transformlike_arg` with `validate_transform4x4`. It  was only used by `Prop3D.user_matrix`. This change broadens the kinds of inputs allowed when setting `user_matrix`. In particular, it allows setting with nested lists.

### CODE REVIEW

File: pyvista/plotting/prop3d.py [hunk 1]
```diff
@@ -370,8 +366,10 @@ def user_matrix(self) -> NumpyArray[float]:  # numpydoc ignore=RT01
         return array_from_vtkmatrix(self.GetUserMatrix())
 
     @user_matrix.setter
-    def user_matrix(self, value: TransformLike):  # numpydoc ignore=GL08
-        array = np.eye(4) if value is None else _coerce_transformlike_arg(value)
+    def user_matrix(
+        self, value: TransformLike | scipy.spatial.transform.Rotation
```
Comment (tkoyama010) [inline pyvista/plotting/prop3d.py:370]: I'd like to make a comment that is a little off-topic from this PR discussion. Should `scipy.spatial.transform.Rotation` be integrated into `TransformLike`? Or does it complicate the code?

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/plotting/prop3d.py (hunk 1 above)
Comment (user27182) [reply pyvista/plotting/prop3d.py:370]: It probably should be part of `TransformLike`.

A related issue is how to deal with the import, similar to https://github.com/pyvista/pyvista/pull/6354#discussion_r1673855824.
Adding it to the aliases in `typing_core` would make it a runtime import, which would also fix the static import issue for `scipy`.

Should we define a new `RotationLike` alias for the 3x3 inputs, and make it a subset of `TransformLike`?

Comment (tkoyama010) [reply pyvista/plotting/prop3d.py:370]: +1 for adding `RotationLike` alias for the 3x3 inputs. If creating a PR not takes your time, then let's create one and see how other developers react.

**Pull request merged**